src/lib/base.jl

```diff
@@ -44,6 +44,27 @@ end
   end
 end
 
+@adjoint function Base._oidd_nextind(a, i)
+  Base._oidd_nextind(a, i), Δ -> begin
+    (nothing, nothing)
+  end
+end
+@adjoint! function get(d::AbstractDict, k, default)
+  hk = Ref{Bool}()
+  val = if haskey(d, k)
+    hk[] = true
+    d[k]
+  else
+    hk[] = false
+    d[k] = default
```

Comment:
`get` should not mutate, maybe you want to define the adjoiint for `get!`?

Reply:
We aren't mutating anything in the user defined objects, so it should be fine. `get!` would still mutate the gradient dictionary so its not any better.

Reply:
you are defining an adjoint for `get(d::AbstractDict, k, default)` that mutates `d`, this is not fine at all

Reply:
Ok, I see what you mean